src/test/run-make/codegen-options-parsing/Makefile

```diff
@@ -22,3 +22,10 @@ all:
 	$(RUSTC) -C lto=foo dummy.rs 2>&1 | \
 		grep 'codegen option `lto` takes no value'
 	$(RUSTC) -C lto dummy.rs
+
+	# Should contain gc-sections...
+	$(RUSTC) -Z print-link-args dummy.rs 2>&1 | \
+	  grep '"-Wl,--gc-sections"'
```

Comment:
This check will probably need to be guarded based on platform as well. I do agree that `objdump` would probably be the best way to go here, but it's probably only reliably available on Linux (and maybe some BSDs). By default it's not on OSX and I doubt it's available on Windows much.

Same with checking for `--gc-sections` as well, unfortunately that's only the name of the argument on Linux. On OSX it's `-Wl,-dead_strip` and on MSVC it's something different as well.